pkg/ccl/backupccl/restore_data_processor.go

```diff
@@ -251,19 +276,35 @@ func (rd *restoreDataProcessor) processRestoreSpanEntry(
 		value.ClearChecksum()
 		value.InitChecksum(key.Key)
 
-		if log.V(3) {
+		if log.V(5) {
 			log.Infof(ctx, "Put %s -> %s", key.Key, value.PrettyPrint())
 		}
+		timeProducing += time.Since(lastAdd)
+		beforeAdd := timeutil.Now()
 		if err := batcher.AddMVCCKey(ctx, key, value.RawBytes); err != nil {
 			return summary, errors.Wrapf(err, "adding to batch: %s -> %s", key, value.PrettyPrint())
 		}
+		timeAdding += timeutil.Since(beforeAdd)
+		lastAdd = timeutil.Now()
```

Comment:
I wouldn't be surprised if these in the hot loop actually slow the loop considerably

Reply:
Removed these. In restore2TB bumping `kv.bulk_ingest.batch_size` avoids size-triggeered flushes during AddMVCCKey, so the time spent in this loop is meaningfully different from the time based flushing below. I think measuring this difference is interesting to see if reading the iterator is taking a substantial amount of time that can be pipelined with the addition of the keys to the batcher.